Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp

```diff
@@ -225,7 +229,17 @@ void webkit_web_view_set_background_color(WebKitWebView* webView, WebKitColor* b
     g_return_if_fail(backgroundColor);
 
     auto& page = webkitWebViewGetPage(webView);
-    page.setBackgroundColor(webkitColorToWebCoreColor(backgroundColor));
+    auto color = webkitColorToWebCoreColor(backgroundColor);
+    page.setBackgroundColor(color);
+#if ENABLE(WPE_PLATFORM)
+    if (auto* view = static_cast<WebKit::PageClientImpl&>(page.pageClient()).wpeView()) {
+        if (color.isOpaque()) {
+            WPERectangle rect { 0, 0, wpe_view_get_width(view), wpe_view_get_height(view) };
+            wpe_view_set_opaque_rectangles(view, &rect, 1);
```

Comment:
So, if I understood well, in general the normal case is setting a window size opaque region due to ["The web view default background color is opaque white, so set the whole view region as opaque initally"](https://github.com/WebKit/WebKit/pull/25344/files#diff-f0d3765a74324105f31da7e43f65412d783bc30f5440777fa09ed47a12d471c0R420)

Reply:
Right